### DISCUSSION TITLE
How to enable a second I2C on the Board to drive OLED-Display?

### DISCUSSION THREAD

**Discussion by ErichHeinemann** [marcel-licence/esp32_midi_sampler#7] [Ideas]
Hi Marcel, I have removed R70 and R68, which are measured by 0Ohm.
That way, I free up Pin 23 and Pin 5 for a second I2C and made the keys 4 and 6 unusable.

My plan was to add a second I2C to implement a simple local UI and add some I2C multiplexer to re-enable the missing pins or to add a rotary encoder.

I am able to drive the OED on these new pins, but after anabling the MIC it crashes. 
I will send You the code tomorrow... perhaps You have time for a review?

As I add some new things and wanted to disable the serial menu (VT100) and the LED-VU-Meter to free up thesee pins to. Finally, the serial-MIDI could be driven by the real Serial to free up the pin 22 for I2C.

Currently the option to disable the LED and the VT100 seems not to be implemented. As I wanted to show the VU-Meter Data in the OLED, I would like to calculatee these values but don´t show them on Your Neopixel-Panel nor in the VT100 but I have to calculate these values.
I have started to implement a lot of #ifdef but I belive, I did it at the wrong places.

Finally, do You see an option to free up Pin 0? It is used in the I2S for external MLCK?

A I2S-Demo with 2 I2C-Bus-Scanners works fine and a Adafruit-Demo for SSD1306-Displays, which I changed for 2 independent I2C-Busses works too.

Best Regards
Erich

**Comment by marcel-licence** [comment]
Hi Erich,
great work!! Sure I am happy to review it.
Some Pins need some handle with care. I also observed using some pins will cause that the automatic firmware download does not work anymore. 
It is a good point to introduce some switches for different features.
Best regards,
Marcel

**Reply by ErichHeinemann** [reply]
I found a way, I removed R70 and E 68 which are both labeled on the board. They were 0 Ohm!? - seems to be a reemovable bridge. Afterwords I was able to activate a Twowire on Pin 5 and 23. Key 6 and Key4 are now disabled but I believe I could rewire them to another I2C-Multiplexer.


#ifdef SSD1306_ENABLED

// #include <SPI.h>
#include <Wire.h>
#include <Adafruit_GFX.h>
#include <Adafruit_SSD1306.h>

#define SDA2 5
#define SCL2 23

//#define SDA1 33 used internally in the ESP32 A1S
//#define SCL1 32 used internally in the ESP32 A1S

// TwoWire I2Cone = TwoWire(0);
TwoWire I2Ctwo = TwoWire(0); // or(1)

#define SCREEN_WIDTH 128 // OLED display width, in pixels
#define SCREEN_HEIGHT 64 // OLED display height, in pixels

#define SCREEN_ADDRESS 0x3C ///< See datasheet for Address; 0x3D for 128x64, 0x3C for 128x32
Adafruit_SSD1306 display(SCREEN_WIDTH, SCREEN_HEIGHT, &I2Ctwo, -1 );
// Some of the Adafruit-Libs have the option to handover a TwoWire-Object


#endif


... later in setup()

#ifdef SSD1306_ENABLED
   // OLED Display
   pinMode( SDA2, INPUT_PULLUP); 
   pinMode( SCL2, INPUT_PULLUP); 
   // I2Cone.begin(SDA1,SCL1,100000); // SDA , SCL of I2C-One are internally
   I2Ctwo.begin(  SDA2,SCL2,100000); // SDA2 pin 17, SCL2 pin 16 
   oled_setup(); // my setup for the OLED ... 
#endif
